src/app/services/idb.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import Dexie from 'dexie';
+import { AnnotationID } from '../models/evt-models';
+
+@Injectable({
+  providedIn: 'root',
+})
+
+export class IdbService extends Dexie {
+  public db: Dexie.Table<AnnotationID, number>;
+  constructor() {
+    super('EVT-Annotator');
+    this.version(1).stores({
+      Annotations: '++id',
+    });
+    this.db = this.table('annotations');
+  }
+
+  getAll() {
+    return this.db.toArray();
+  }
+
+  add(data) {
+    return this.db.add(data);
+  }
+
+  update(id, data) {
+    return this.db.update(id, data);
+  }
+
+  remove(id) {
+    return this.db.delete(id);
+  }
```

Comment:
Why do we need to wrap db?
Do you expect the use of Dexie to change over time?

Reply:
I remove them and I use the functions of the DB directly.